JUnit5-README.md

```diff
@@ -0,0 +1,193 @@
+
+### Describe cloud deployment
+You can describe resources to be deployed using annotations `@With*` on a test class. See following examples for Azure/AWS specific resources or JClouds that provide abstraction for several Cloud resources.
+
+#### Azure templates
+
+```java
+@Sunstone
+@WithAzureArmTemplate("appservices/clusteredWebApp.json")
+public class AzureTest {}
+```
+
+In this case, template is located in resources. You can also provide a raw content or URL and the template shall be downloaded. Resources are deployed to the group and region defined in `sunstone.properties`.
+
+You can download JSON definition of already existing resources or downloading JSON for automation in create form.
+
+Azure template allows you to deploy and configure Azure specific resources using native technologies.
+
+#### AWS CloudFormation template
+
+```java
+@Sunstone
+@WithAwsCfTemplate("aws/messaging-mdb.yaml")
+public class AWSTest {}
+```
+
+In this case, template is located in resources. You can also provide a raw content or URL and the template shall be downloaded. Resources are deployed to a stack of random name in region defined in `sunstone.properties`
+
+You can use Former2 tool to obtain YAML definition of already existing resources.
+
+CloudFormation template allows you to deploy and configure AWS specific resources using native technologies.
+
+#### JClouds
+TODO
+
+```java
+@Sunstone
+@WithJClouds("test.properties")
+public class JCloudTest {}
+```
+
+In this case, property file is located in resources. You can also provide a raw content or URL and the file shall be downloaded.
+
+Contract for different providers can be found at:
+- [azure-arm-README](azure-arm-README.md)
+- [baremetal-README](baremetal-README.md)
+- [ec2-README](ec2-README.md)
+- [docker-README](docker-README.md)
+- [openstack-README](openstack-README.md)
+- [wildfly-README](wildfly-README.md)
+
+JCloud enables you to share a test among different providers.
+
+### Provide additional configuration
+Sunstone allows you to define a setup task with enabled DI. Add the `@Setup` annotation on test class
+
+```java
+@Sunstone
+@WithAwsCfTemplate("aws/messaging-mdb.yaml")
+@Setup(Configure2LCApp.class)
+public class AWSTest {}
+```
+```java
+
+public class Configure2LCApp extends AbstractSetupTask {
+    
+  @SunstoneResource(of = "eapqe-instance", hint = AWS_EC2_INSTANCE)
+  OnlineManagementClient client;
+
+  @Override
+  public void setup() throws Exception {
+    // configure eap
+  }
+
+  @Override
+  public void cleanup() throws Exception {
+    // clean the configuration
+  }
+}
+```
+
+
+### Use DI for test development
+Sunstone allows you to inject several resources: EAP management client, cloud clients, ... Fields may be either static (injected once) or non-static (injected per test case).
+
+```java
+@Sunstone
+@WithAwsCfTemplate("aws/messaging-mdb.yaml")
+@Setup(Configure2LCApp.class)
+public class AWSTest {
+  @SunstoneResource(of = "eapqe-mdb", hint = AWS_EC2_INSTANCE)
```

Comment:
I think that params for this annotaion could better desribe what is meant here. `of` is actually VM instance name but I had to look at code to figure it out. `hint` would be great to get rid of and somehow detect the cloud provider. I'm not sure if it's feasable.

Reply:
Sunstone can guess a cloud, but can't determine resource type just by the name. In AWS unique name is not required, not even for EC2 instance. I can have two of the same name, I can have a load balancer of the same name, so we need to differentiate between resource types. 

I chose hint, because enum can be easily re-used for `@WildFlyDeployment` and other annotations.

`of` is tricky, what about `identified=` or `resource=`? I have deviated from `named` because it is IMO strongly bound to some kind of a name, which may not be accurate and some resources may not have a "name".